### ISSUE TITLE
Is a WebExtensions version  for Firefox 57+ in the works?

### ISSUE THREAD

**Issue by alerque** [docwhat/itsalltext#94]
I see a note in the README about a version 2 that sounds like it might be related to the development needed to switch to WebExtensions, but the last work on the repository listed is from 5 years ago.

Is there any prospect of a FF57+ version on the horizon?

**Comment by tschwinge** [comment]
Confirming that version 1.9.3 no longer works with Firefox 57 (as provided by Debian experimental, 57.0~b6-1).  <https://support.mozilla.org/en-US/kb/firefox-add-technology-modernizing>, <https://developer.mozilla.org/en-US/Add-ons/Working_with_multiprocess_Firefox>.

**Comment by l-mb** [comment]
I'd also love to see this. Couldn't find an alternative extension that does the same.

**Comment by anarcat** [comment]
this was thoroughly discussed in #70 - the takeaway is that this extension needs to be re-written from scratch and, because of the new web-ext sandbox design, it will be much harder to implement - which is a shame - at this point, what will happen is that this extension will simply die.

alternatives may be the existing [chromium extensions](https://superuser.com/questions/261689/its-all-text-for-chrome) that do similar things - but those require a server to run on the other end for the extension to talk to. it's super clunky...

i've looked briefly at what would be involved in implementing this from scratch, and it *does* seem possible to do things. the basic problem to solve is [working with local files](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Working_with_files#Process_files_in_a_local_app) where it is suggested to use *Connection-based messaging* (the client/server network-port based approached i was refering to above) or *Connectionless messaging* which delivers a JSON message to a temporary app which talks to your app.

A [sample extension](https://github.com/mdn/webextensions-examples/tree/master/native-messaging) is available in the demo that does a simple ping/pong with a Python program. 

Then there's this hilarious [porting guide](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Porting_a_legacy_Firefox_add-on) which can be summarized as such:

![just draw the rest of the fucking owl](https://user-images.githubusercontent.com/796623/31549760-26912d70-affd-11e7-9940-6281b5e2e182.png)

so basically, i think we're stuck. i would suggest people to put their efforts on the [emacs chrome](https://github.com/stsquad/emacs_chrome/) extension for now, which is being [ported to Firefox](https://github.com/stsquad/emacs_chrome/pull/130). another option is [GhostText](https://github.com/GhostText/GhostText) which seems to be a [working alternative](https://github.com/GhostText/GhostText/issues/25) for *both* firefox and Chrome and that supports more than just emacs (through editor-specific plugins). i had good success making GhostText work, except some [silent errors with umatrix](https://github.com/alpha22jp/atomic-chrome/issues/27) which are easily fixed.

good luck!

**Comment by docwhat** [comment]
> i would suggest people to put their efforts on the [emacs chrome](https://github.com/stsquad/emacs_chrome/) extension for now, which is being [ported to Firefox](https://github.com/stsquad/emacs_chrome/pull/130). another option is [GhostText](https://github.com/GhostText/GhostText) which seems to be a [working alternative](https://github.com/GhostText/GhostText/issues/25) for *both* firefox and Chrome and that supports more than just emacs (through editor-specific plugins).

Yeah, I think IAT is at the end of its life.  RIP old friend.

**Comment by alerque** [comment]
That sound semi official. I'll close this issue as it seems to be resolved.

Thanks for all the fish! I used this extension for a long time and (paired with Neovim in a popup terminal) wrote a lot of text through it.

For my part I've started looking into GhostText. A little rough on the setup end but it looks promising.

**Comment by docwhat** [comment]
Yeah, I updated the README... IAT died on October 31st, 2017 — Halloween.

**Comment by stiv-sigmal** [comment]
It doesn't seems like emacs_chrome is alive. So the only alternative is GhostText — the very unusual (uncomfortable) extention for external editing of textareas. To use it, I must run text editor manually, click on button, then click on textarea and so on. I can't save text to edit in next time without some excess steps with clipboard and so on. In that way I just can use clipboard without any additional extentions.

**Comment by gibus** [comment]
After having bad experiences with some other alternative (GhostText, Textern), I'm using withExEditor for a couple of days and eventually found it gives same functionalities as IAT.

You have to install a nodejs host (see https://github.com/asamuzaK/withExEditorHost. The only difficulty is to have nodejs > v8.9.0 (for Debian Stretch or Ubuntu Xenial, you've got to installing it from https://nodejs.org/en/download/package-manager/).

You've got an install and configure scripts to run and then the host is transparently launched by Firefox, and a right click on any textarea gives allows you to "edit with XXX".

**Comment by dakra** [comment]
FYI: I just uploaded the firefox version of edit-with-emacs to AMO: https://addons.mozilla.org/en-US/firefox/addon/edit-with-emacs1/

You can use other editors with the script from the github homepage but with emacs it's very easy. Just install the `edit-server` package and put `(edit-server-start)` in your config.
